### ISSUE TITLE
Where is the ubuntu version?

### ISSUE THREAD

**Issue by jkm9000** [nylas/nylas-mail#3256]
Wanted to check this out but I'm unable to locate any sort of download for Ubuntu (or any other Linux variant).

The README (https://github.com/nylas/nylas-mail/blob/master/README.md) states "You can download compiled versions of Nylas Mail for Windows, Mac OS X, and Linux (.deb) from 
https://nylas.com/download" 

Yet when I visit that link and click Linux you ask for my email address with a message that it is coming soon.

This page (https://nylas.github.io/nylas-mail/) also has a link to download for Linux which points to https://edgehill.nylas.com/download?platform=linux

When I visit that the page displays "No matching releases"

Does it exist and I'm missing it?

**Comment by unsalted** [comment]
I can confirm there used to be one because I have it on my laptop.  Went to add it to Desktop and it wasn't there.

**Comment by ronisbr** [comment]
The last time I checked, they were preparing a linux/windows build for the new Nylas Mail (Free). However, this was 3 weeks ago. I haven't heard any news after that.

**Comment by jkm9000** [comment]
Thanks folks. Only just learning about this app, it's quite confusing that there obviously was a linux version and now there is not. Especially when all the docs I've seen state it exists now.

**Comment by unsalted** [comment]
There was one because like I said I have it running on Ubuntu on my other laptop.  Some clarification would go a long way.

**Comment by ArchangeGabriel** [comment]
You’re confused with N1 vs Nylas Mail, which is kinda normal.

Nylas had one product for a long time, the N1 mail client. This one was (and is still) available for Linux and Windows, but is only usable with the Nylas Pro paid accounts. You can grab it from there: https://billing.nylas.com/download-pro.

Recently, Nylas introduced Nylas Basic accounts, which are free, but only work with the new Nylas Mail app. This one is not yet available for Linux nor Windows (only MacOS at this time).

In the future, what will happen is:
1. Nylas Mail will be available for Linux and Windows.
2. Nylas N1 will be removed in favour of Nylas Mail, which would then support both paid accounts (Nylas Pro tier) and free ones (Nylas Basic).

**Comment by kirbyfan64** [comment]
@ArchangeGabriel Is there roughly an ETA when the Linux and Windows versions will be available?

**Comment by jacobmischka** [comment]
Not sure, but I just noticed that if you click Windows or Linux on [the download page](https://billing.nylas.com/download) it'll let you enter your email and they'll notify you when it's finished.

I checked this page this morning, and Windows was listed but Linux was not. So at the very least this is actively being kept track of it seems.

**Comment by ArchangeGabriel** [comment]
@kirbyfan64 “They’ll be ready in a couple weeks“ (from the [announcement blog post](https://blog.nylas.com/nylas-mail-is-now-free-8350d6a1044d)).

**Comment by ronisbr** [comment]
Does anyone have any news after that blog post?

**Comment by ArchangeGabriel** [comment]
From the mail sent on for 1.0.18 release, the builds were coming “very soon”.

**Comment by jtflynnz** [comment]
From the mailer announcing Nylas-Mail updates on 2017-1-31: 

> PS: Windows and Linux builds are coming very soon. Thanks for your patience!

**Comment by jonsparks** [comment]
Any news or an ETA on the Linux version? The README still says a .deb is available at nylas.com/download, but there's not one- it might be a good idea to update that section to say it's coming soon.

**Comment by ArchangeGabriel** [comment]
Hum, that section of the README just got name changes, but everything is still relative to N1 (Windows/Linux builds, Arch packages…).

**Comment by ronisbr** [comment]
This message is from @GM-Polyakov that was posted to #3294 

> I don't have a public ETA I can share, but I can tell you it's on the order of weeks, not months.

**Comment by Tobishua** [comment]
It's really strange because Nylas can't make Linux and Windows version for a long time... But this is Electron.js so what the problem? :cry:

**Comment by sachinag** [comment]
So! The problem is that, sure, it's cross platform, but there are still platform specific bugs we need to work out. I promise promise promise we're working on it and we expect it to be weeks, not months. 

Pro (old N1) still has Linux available at nylas.com/download-pro ; we know this has been frustrating and we're sorry we weren't able to push the Linux version of the new Nylas Mail out as fast as we wanted to (and said we would).

**Comment by bithooked** [comment]
I guess I don't understand this approach.  What is the point of a FOSS GPL product, if you don't develop it in the open, warts and all?  Most of us Linux users are used to dealing with bugs, and most of us here on Github would rather have a chance to wack those bugs for you than be in the dark.  I'm not saying you put it on your main download site, but keeping it completely proprietary and in the dark seems backwards.

**Comment by jtflynnz** [comment]
Just to clear up any misinformation I have spread (in the issue 3315 from above, for instance), one can still build the sync-engine and self-host under the old subscription model, if you want to use it without paying for Pro. See docs reference here: https://github.com/nylas/nylas-mail/blob/master/CONFIGURATION.md

**Comment by jtflynnz** [comment]
Correct me if I'm wrong, but isn't what's posted here the latest source code? Including what would be used to build the Linux client? This wouldn't be "closed source"/"proprietary" if the code is truly posted here still (where it is also still under the same license and can be compiled), it would just not be compiled into a Linux format yet (and there don't appear to be instructions for compiling yet).

Also FWIW, while I think an opt-in, build it yourself beta test group would still benefit the new Nylas-Mail, I also understand where the team is coming from.  There are many projects that have hurt their momentum by pushing out a half-baked product too early, and "open" doesn't have to mean access to buggy and/or beta versions before they feel ready (as long as source code is available).

Again, all said and done, I think they may benefit from additional testers and making it easier to opt-in to test versions, but as long as the source is available, I can live with it until they are ready (Here's hoping we're getting close!)

In that vein, there do appear to be updates to some of the docs over the last few months referencing build requirements, so maybe they are getting closer to announcing that? 

See here (a little old now, but still something): https://github.com/nylas/nylas-mail/commit/2170926dc7faea8d524c3139a11c3d33625cb418 and https://github.com/nylas/nylas-mail/blob/master/.travis.yml#L10

Perhaps someone may be able to figure out how to build the new version?

**Comment by jacobmischka** [comment]
As far as I know the current source in this repo only will work if you're a pro subscriber or self host the sync engine, the free tier requires some pieces that haven't been open sourced.

**Comment by jacobmischka** [comment]
Oh I didn't realize that you're the same person who just posted that half an hour ago, you know more than I do then. I didn't know until you said so that you could still self host under the new pricing model, so thanks for clearing that up!

**Comment by bithooked** [comment]
Perhaps I'm mistaken.  I thought I read elsewhere that this project can't be built on Linux either, but maybe that's not current state.  For someone who's yet to experience Nylas (because I can't download it for my platform) it's really hard to even understand the phrase "build the sync-engine and self-host under the old subscription model, if you want to use it without paying for Pro".

**Comment by jacobmischka** [comment]
Prior to the recent introduction of "Nylas Mail", the same product was released under the name "Nylas N1". The client was pretty similar, but it synced from the Nylas Mail Engine, which in turn pulled from your email accounts. For a time N1 was available for free, but it currently requires a subscription to pull mail using their Sync Engine.

Instead of doing that, you can compile and run an instance of the Sync Engine yourself, as it is also open source (I believe it's missing some features which are still closed source, however). If you do this, you can instruct the N1 client to sync from there instead of the Nylas-hosted engine. Nylas Mail is essentially N1 + Sync Engine in a single product, which is easier than managing the sync engine and the client separately.

**Comment by ArchangeGabriel** [comment]
@bithooked Open-Source software doesn’t imply open-source dev at all stages. They accept PR, you’re free to fork, but the dev head, design decision and such are their owns. Also, @jtflynnz it seems to me that commits are only landing on GitHub from time to time, probably when release happen, so current master is not latest source code (but likely latest distributed binaries source code — excepted for the proprietary part obviously). I can name you at least one big example of open-source software pushing this dev model to the extreme: Android.

Now, regarding compiling N1 or Mail, AFAIK in the current state both require access to a private submodule (can’t find better than [here](https://github.com/nylas/nylas-mail/issues/3270#issuecomment-279005111) for Mail; the module name is K2 for N1), so that both are not buildable from source. Now I haven’t dig at all into that and maybe K2 is not needed for most features and you can build Nylas N1 without it and make it work with a self-hosted engine. Or maybe you can compile it at a former point where it didn’t depend on K2.

So if the development model isn’t violating GPL at all, the situation with this private submodule while they are distributed binaries is a bit more shady. But Nylas violating (hopefully in a temporary manner) their own license is kinda, you know… hard to make something of.

**Comment by bithooked** [comment]
@jacobmischka Got it, thanks.  That makes a lot more sense.  With that information in hand, I'll wander off and start cloning and building, and see what I can make of it.

**Comment by kirbyfan64** [comment]
@bithooked

> **building**, and see what I can **make** of it

(FWIW, I managed to get it to build, but the source itself requires a local sync server.)

**Comment by ArchangeGabriel** [comment]
So, I was misremembering but finally came across my post regarding those submodules: https://github.com/nylas/nylas-mail/issues/3244#issuecomment-281375121

So K2 is the missing one for Mail, and there seems to be an `edgehill` thing for N1.

**Comment by jtflynnz** [comment]
As an update: builds appear to be out for testing. I removed links, in case it's not supposed to be posted (but just signup for the newsletters to get them!), bu this is from the latest email (today):

We’re incredibly excited to announce the next generation of Nylas email: Nylas Mail 2.0. As a valued Nylas N1 customer we wanted you to have the chance to try Nylas Mail and to give us feedback. Nylas Mail has all of the most popular features of Nylas N1 and we have added additional security and performance capabilities.  Nylas Mail includes:  Unified Inbox, Tracking, Send Later, Reminders, and much more.  Nylas Mail is available on Windows,  MacOS and Linux.  Additionally, unlike N1 all of your emails are stored on your computer and we do not keep a copy of the emails on our secure Nylas cloud. 

All of your accounts should work with Nylas Mail because they are Office365 compatible.  To help with this we’ve written up a short Knowledge Base Article on how to connect outlook.com, hotmail.com, or live.com accounts to Nylas Mail.

Our goal is help all of our Nylas N1 customers move to Nylas Mail in the next month.  In order to make the transition easier, we first want to make sure it’s working well for our existing users including you. 

OSX:  [Removed]
Win:   [Removed]
Linux: [Removed]

Please reply to me directly with your thoughts, comments and suggestions.

Thank you for supporting us!

-Michael

**Comment by jacobmischka** [comment]
Based on the text in that email it looks like that was sent only to Nylas Pro subscribers, I believe I'm subscribed to the newsletter and I didn't get anything.

Anyway, excited that it looks like it'll be out soon!

**Comment by tomhoogeveentje** [comment]
It seems that they have arrived!
https://billing.nylas.com/download

**Comment by almet** [comment]
Thanks! I just installed it and it works great :-)

**Comment by ArchangeGabriel** [comment]
2.x is indeed officially released now (https://github.com/nylas/nylas-mail/releases/tag/2.0.10), although the corresponding source code is not (the tarball attached there is the same as for all releases since 1.0.29, since this is what the GitHub repo contains currently).

So no building from source currently.

**Comment by ronisbr** [comment]
Just to be sure, now we need to have a Nylas ID but our mail will not be synced on their servers anymore right?

Furthermore, if I decide to buy nylas pro (some features are interesting), then will they sync my mails to their servers?

**Comment by ArchangeGabriel** [comment]
Nylas ID shouldn’t be mandatory. FWIU that was one of the goal, but maybe they did not reach it yet. At least it seems from the changelog that you can sign out of your NylasID/not have one (from the 2.0.1 release):
> Make sure app can update even after signing out of NylasID
> Don't make any requests when NylasID isn't present

Now regarding Nylas Pro, FWIU, no, they will not sync your emails to their servers. Remind this sentence from @jtflynnz received email:
>Additionally, unlike N1 all of your emails are stored on your computer and we do not keep a copy of the emails on our secure Nylas cloud.

That being said, I think that at least in one case your email will have to go through their server: send later. Because what if your system is not on at the time of sending? So I guess that how it works is put your to-be-send email on their server with the expected send date/time, and then send it at this given point. Unless they changed the behaviour to send ASAP once that date/time is over, in which case it could be done client side. Both cases could be implied by this changeline, depending on what limits mean here:
> Add feature limits to reminders and send later

Anyway, I expect Nylas to do a proper announcement/blog post of the current state with this 2.0 release, especially regarding the following points:

– is a server-side component still needed currently, for what services, and where are you on the path to make it optional?
– quite linked to this first one, what goes through the server-side component?
– what are now the features differences between Free/Paid (Basic/Pro)? The pricing page has changed somewhere between Nylas Mail and Basic announcement and now, so that it does not say anything about this anymore. For instance, it looks like O365/Exchange is not in the “developer” version, but then either @jtflynnz has a different pricing tier or this is not the case. Also, I would expect things like “mail merge, tracking, and dozens of premium features including our Salesforce CRM integration” (from https://blog.nylas.com/nylas-mail-is-now-free-8350d6a1044d announcement) to appear on this page (btw, the Salesforce CRM link brings to the homepage now).
– can we still self-host the server-side component, what does it give us/how does that differ from Pro?

**Comment by MichaelTunnell** [comment]
I just tried to use the DEB file from the download page . . .

1. Required me to create a Nylas ID (I stopped here and refused to continue)
2. When installing the terminal output said I wasn't using Ubuntu (which I am)
> You are not running Debian, Ubuntu, ElementaryOS or LinuxMint.  Not adding Nylas repository.

I don't mind telling you that I am using your software . . . I mind that you require me to.

**Comment by Isotop7** [comment]
So there is no possibility to use Nylas mail at the moment without creating or having a Nylas ID?

I dont see any option to use the basic version. After going through 'Getting started' I get a prompt so sign in.

https://www.howtoforge.com/tutorial/how-to-set-up-and-use-nylas-n1-email-client/

**Comment by ronisbr** [comment]
I'm using it on my openSUSE tumbleweed without problems. I read the terms that you agree before creating an account and it seems that the part that authorized Nylas to store your emails has been removed (I am not a lawyer though). So, if by creating this account will save my setup between my computers, then I am glad to do so!

As of now, I saw no problems at all! Nylas is amazing, for sure the most modern mail client available for Linux. I still miss somethings like plain text mail composition, but it seems to going to the right direction. Congratulations for the team!

**Comment by kirbyfan64** [comment]
@Isotop7

> After going through 'Getting started' I get a prompt so sign in.

You can create a free Nylas account.

**Comment by alexanderadam** [comment]
So the current state got a bit worse than with Nylas N1. With Nylas N1 you were able to host the sync engine on your local machine or own server. Which means you just had to deal with the [privacy issues of being tracking with your personaly emails by another party](https://github.com/nylas/nylas-mail/issues/1849#issuecomment-293896007).

And now it seems that you can't use your own infrastructure anymore and you need to have a Nylas account.

**Comment by Isotop7** [comment]
@kirbyfan64 Yes, but i dont have the possibillity to use it without an account?

**Comment by MichaelTunnell** [comment]
> You can create a free Nylas account.

The issue is not that you can or can't but that you **have to**. I don't want them to have any of my data including which accounts are associated to me. If I could use it without an ID then even if they tracked every account logged in, they wouldn't know who exactly connected them. With this, they have more metadata than I personally ok with them having.

Not to mention the $99-$1299 per month for more than 10 accounts is insane.

**Comment by Karlsoost** [comment]
Any update on this?

Thanks,
Heiko

**Comment by kirbyfan64** [comment]
@Karlsoost It already came out several days ago...

**Comment by jstejada** [comment]
Correct, you can download it from nylas.com/nylas-mail
